karate-core/src/main/java/com/intuit/karate/core/FeatureParser.java

```diff
@@ -81,15 +82,29 @@ public static Feature parseText(Feature old, String text) {
         return feature;
     }
     
-    public static void updateStepFromText(Step step, String text) {
+    public static boolean updateStepFromText(Step step, String text) {
         Feature feature = new Feature(step.getFeature().getResource());
+        final List<String> prefix = Arrays.asList("*","Given","When","Then","And","But");
+        final String stepText = text;
+        boolean hasPrefix = prefix.stream().anyMatch(prefixValue -> stepText.trim().startsWith(prefixValue));
+        // to avoid parser considering text without prefix as Scenario comments/Doc-string
+        if(!hasPrefix){
+        	return false;
+        }
         text = "Feature:\nScenario:\n" + text;
-        feature = new FeatureParser(feature, FileUtils.toInputStream(text)).feature;
-        Step temp = feature.getStep(0, -1, 0);
-        step.setPrefix(temp.getPrefix());
-        step.setText(temp.getText());
-        step.setDocString(temp.getDocString());
-        step.setTable(temp.getTable());
+        FeatureParser fp = new FeatureParser(feature, FileUtils.toInputStream(text));
+        if(!fp.errorListener.isFail()) {
+        	feature = fp.feature;
+            Step temp = feature.getStep(0, -1, 0);
+        	if(temp != null) {
```

Comment:
please have a space between if and the (

Reply:
done